merlin_standard_lib/utils/misc_utils.py

```diff
@@ -236,4 +236,42 @@ def validate_dataset(paths_or_dataset, batch_size, buffer_size, engine, reader_k
             reader_kwargs["batch_size"] = buffer_size
     else:
         reader_kwargs["part_mem_fraction"] = buffer_size
+
     return Dataset(files, engine=engine, **reader_kwargs)
+
+
+def _augment_schema(
```

Comment:
In the T4Rec data loader, we have a filtering step happening [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/776af7bad969fb408e623d1a3b832e8d5ee337a4/transformers4rec/torch/utils/data_utils.py#L228) to filter out the columns that are not used by the model before passing the dataset to the loader. This step is required to make sure that the data loader is returning only the input features required by the model  . 

so perhaps we can add this filtering step to the `_augment_schema` method to keep all the schema processing logic in one place.